geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/executor/hash/HScanExecutor.java

```diff
@@ -67,12 +67,13 @@ public RedisResponse executeCommand(Command command,
     }
 
     RedisKey key = command.getKey();
-    if (!getDataRegion(context).containsKey(key)) {
+    RedisData value = context.getRegionProvider().getRedisData(key, null);
```

Comment:
you could use "getRedisData(key)" to set value and then "value.isNull()" instead of "value == null".